app/src/main/java/com/android/androidfundamentalsgroup1/MainActivity.java

```diff
@@ -59,8 +77,38 @@ protected void onCreate(Bundle savedInstanceState) {
         // setupSpinnerAdapter();
 
         // run RecyclerView sample
-        setContentView(R.layout.views_sample_recycler_view);
-        displayEmailsList();
+        // setContentView(R.layout.views_sample_recycler_view);
+        //  displayEmailsList();
+    }
+
+    /* Function that checks if email and phone fields are completed (and valid) and checkbox is checked */
+    public void submit(View view) {
+        String phone = editTextPhone.getText().toString();
+        String email = editTextEmail.getText().toString();
+        checkBoxTermsAndConditions = findViewById(R.id.checkBoxTermsAndConditions);
```

Comment:
Please consider getting the checkbox inside of the onCreate method or define a dedicated method for this purpose and call it in onCreate. Now if you don't press the button and want to access the checkbox it will crash with NullPointerException